### ISSUE TITLE
Automatically enable numbering of equations that have a label

### ISSUE THREAD

**Issue by nikhilweee** [typst/typst#3031]
### Description

Right now if I label an equation when numbering is turned off, I get an error.
```typst
$ P(A|B) = P(A,B) / P(B) $ <conditional>

@conditional defines conditional probability
```
```
cannot reference equation without numbering
Hint: you can enable equation numbering with `#set math.equation(numbering: "1.")`
```

That's great, but it would be cool to automatically number those equations for me instead.

### Use Case

This can be particularly useful when a document has multiple equations but not all of them need to be numbered. If this feature is implemented, only the ones that have a label will be numbered, resulting in a cleaner output.

**Comment by OrangeX4** [comment]
Currently you can use the i-figured package to do this. Maybe there will be built-in support for typst in the future.

**Comment by Enivex** [comment]
I'll close this in favor of https://github.com/typst/typst/issues/380

Note that there's also a forge on the discord server about equation numbering

**Comment by eltos** [comment]
For discoverability:

Multiple solutions using show rules can be found in this article in the Typst forum:
https://forum.typst.app/t/how-to-conditionally-enable-equation-numbering-for-labeled-equations/977/14

**TL;DR**
Add this to your document:
```typst
#set math.equation(numbering: "(1)")
#show math.equation: it => {
  if it.block and not it.has("label") {
    counter(math.equation).update(v => v - 1)
    it.body
  } else {
    it
  }  
}
```

**Comment by angeris** [comment]
Not to revive a dead proposal, but it would be really quite nice to have this since having _every_ equation labeled (even if it's never referenced) is a very weird default when writing papers!

While the proposed solution from https://forum.typst.app/t/how-to-conditionally-enable-equation-numbering-for-labeled-equations/977/14 is good enough, I think it is better served by, at least, having an option that can be set conditionally.

**Comment by laurmaedje** [comment]
@angeris This is a duplicate issue. The main issue (https://github.com/typst/typst/issues/380) is still open.

**Comment by angeris** [comment]
got it, thanks !

EDIT: wait... this doesn't look quite the same unless I'm missing something.

**Comment by laurmaedje** [comment]
The top level issue description discusses just one thing, but the ongoing discussion touches on many more aspects which need to be considered holistically, including yours.

**Comment by angeris** [comment]
right, ok, maybe I'm going crazy (which is fair!) I just don't see where this appears anywhere in the discussion.

I see things like subnumbering, referencing specific subequations in a block, prefixing equation numbers by sections (etc), but at no point in the discussion do I see anything resembling "I would like a flag to have equations default to being unnumbered unless tagged"?

(unless I missed it, which is very possible)

**Comment by Enivex** [comment]
> right, ok, maybe I'm going crazy (which is fair!) I just don't see where this appears anywhere in the discussion.
> 
> I see things like subnumbering, referencing specific subequations in a block, prefixing equation numbers by sections (etc), but at no point in the discussion do I see anything resembling "I would like a flag to have equations default to being unnumbered unless tagged"?
> 
> (unless I missed it, which is very possible)

It's mentioned in a comment that's linked to in that issue, and has also come up on Discord.

**Comment by laurmaedje** [comment]
> right, ok, maybe I'm going crazy

Sorry for sending you on a ghost hunt. I was sure it was there because it's been discussed at various times, but maybe that was on Discord. I still think the more general issue makes sense because everything needs to be designed together, but I guess we can use the subissue feature for this.

**Comment by angeris** [comment]
got it! that's fair I didn't go into the Discord discussion, so might have been there!

Thanks @laurmaedje ! :)
